python/pyspark/errors/error_classes.py

```diff
@@ -347,6 +347,11 @@
       "Index must be positive, got '<index>'."
     ]
   },
+  "INDEX_NOT_ZERO": {
```

Comment:
The error class should tell about the issue not solution, IMHO. Could you rename it to: `ZERO_INDEX`